third-party/cmake/LibvtermCMakeLists.txt

```diff
@@ -60,13 +68,24 @@ set_target_properties(harness PROPERTIES
   # run-test.pl expects to find the harness in t/.libs/
   RUNTIME_OUTPUT_DIRECTORY t/.libs)
 
-file(GLOB TESTFILES ${CMAKE_SOURCE_DIR}/t/*.test)
-add_custom_target(check)
-foreach(testfile ${TESTFILES})
-  get_filename_component(target_name ${testfile} NAME_WE)
-  add_custom_target(${target_name}
-    COMMAND ${PERL_EXECUTABLE} ${CMAKE_SOURCE_DIR}/t/run-test.pl ${testfile}
-    COMMENT "**${target_name} **"
-    DEPENDS harness)
-  add_dependencies(check ${target_name})
-endforeach()
+if(Perl_FOUND)
+  file(GLOB TESTFILES ${CMAKE_SOURCE_DIR}/t/*.test)
+  add_custom_target(check)
+  foreach(testfile ${TESTFILES})
+    get_filename_component(target_name ${testfile} NAME_WE)
+    add_custom_target(${target_name}
+      COMMAND ${PERL_EXECUTABLE} ${CMAKE_SOURCE_DIR}/t/run-test.pl ${testfile}
+      COMMENT "**${target_name} **"
+      DEPENDS harness)
+    add_dependencies(check ${target_name})
+  endforeach()
+
+  foreach(header_path ${TBL_FILES_HEADERS})
+    get_filename_component(header_name ${header_path} NAME)
+    set(perl_header_path ${CMAKE_BINARY_DIR}/encoding-test/${header_name})
+    add_custom_target(test-${header_name}
+      COMMAND ${CMAKE_COMMAND} -E compare_files
+              ${header_path} ${perl_header_path}
+      DEPENDS ${header_path} ${perl_header_path})
+  endforeach()
```

Comment:
How does one actually run these? Perhaps they should be dependencies on one of the main targets (`lint` or `test` ?) Otherwise this code is "dead", no one will know to run it :) I guess the `check` target above was never run either.

Reply:
I was running them using `cmake --build . --target test-{name}.inc`. I decided to use `add_custom_target` for consistency with the `check` target, but CMake also has the `add_test` command. To run those tests, the `ctest` tool would need to be used. If you want them to run during every build then adding them all as a dependencies of one main target is probably the best way of doing it.